minitwit_admin/src/compositionStore/index.js

```diff
@@ -1,22 +1,45 @@
-import { readonly, reactive } from 'vue'
+import { provide, inject, reactive, computed } from 'vue'
+import sidebar from '@/compositionStore/sidebar/sidebarModule'
 import async from '@/compositionStore/async/asyncModule'
 import twits from '@/compositionStore/twits/twitsModule'
 import users from '@/compositionStore/users/usersModule'
 import followers from '@/compositionStore/followers/followersModule'
 
-const state = reactive({
-    isSidebarMinimized: false
-})
+const initStore = () => {
+    provide("useSidebar", sidebar)
+    provide("useAsync", async)
+    provide("useTwits", twits)
+    provide("useUsers", users)
+    provide("useFollowers", followers)
+    provide("useSidebar", sidebar)
+}
+
+const useAsync = () => inject("useAsync")
+const useTwits = () => inject("useTwits")
+const useUsers = () => inject("useUsers")
+const useFollowers = () => inject("useFollowers")
+const useSidebar = () => inject("useSidebar")
 
-const actions = {
-    toggleSidebar: () => {
-        state.isSidebarMinimized = !state.isSidebarMinimized
-    }
+export { 
+    initStore,
+    useSidebar,
+    useAsync,
+    useTwits,
+    useUsers,
+    useFollowers,
+    async,
+    twits,
+    users,
+    followers
```

Comment:
I don't think we need to export anything from index.js. The provide/inject is supposed to ease passing components from parent to distant children so I guess if we are calling initStore() on App.vue level, each other vue component embedded in our app will be able to retrieve the provided value on its own level (fx. UserForm in setup() can do inject("useUsers") to get access to the methods).

TLDR: I think its a bit contraintuitive to provide and inject on the same level (ref: https://vuejs.org/guide/components/provide-inject.html#inject).

Reply:
Yes, they are there because I did not want to break existing functionality, there are components still using them.
And by importing useSomething() you dont have to use inject("useSomething") in the setup method, I think its more convenient that way. UserProfileScreen.vue is a good example

Reply:
alrighty then, just making sure its a conscious choice